sentry/src/routes/channel.rs

```diff
@@ -228,3 +234,139 @@ pub async fn create_validator_messages<A: Adapter + 'static>(
         }
     }
 }
+
+// TODO: Maybe use a different error for helper functions like in campaign routes/move this somewhere else?
+async fn create_spendable_document<A: Adapter>(adapter: &A, pool: DbPool, channel: &ChannelV5, spender: &Address) -> Result<Spendable, ResponseError> {
+    let deposit = adapter.get_deposit(&channel, &spender).await?;
+    let spendable = Spendable {
+        channel: channel.clone(),
+        deposit: Deposit {
+            total: UnifiedNum::from_u64(deposit.total.to_u64().expect("Todo")),
+            still_on_create2: UnifiedNum::from_u64(deposit.still_on_create2.to_u64().expect("Todo")),
+        },
+        spender: spender.clone(),
+    };
+
+    // Insert latest spendable in DB
+    insert_spendable(pool, &spendable).await?;
+
+    // TODO: Check and handle the case if an error is encountered after the insertion (ex: During total_spent calculations)
+
+    Ok(spendable)
+}
+
+pub async fn get_spender_limits<A: Adapter + 'static>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let route_params = req
+        .extensions()
+        .get::<RouteParams>()
+        .expect("request should have route params");
+
+    let channel = req
+        .extensions()
+        .get::<ChannelV5>()
+        .expect("Request should have Channel")
+        .to_owned();
+
+    let channel_id = ChannelId::from_hex(route_params.index(0))?;
+    let spender = Address::from_str(&route_params.index(1))?;
+
+    let default_response = Response::builder()
```

Comment:
Since all values are needed for the spender_leaf (which is an Option) you can instead match both values:

- fetch approve state
- fetch new state
- fetch the new_state.balances.spenders.get(spender) = total_spent

If any of the values is not found (i.e. is `None`), return None for `spender_leaf`